api/modules/budgeting.js

```diff
@@ -1,9 +1,33 @@
 const moment = require('moment')
+const { DEFAULT_STRIPE_CURRENCY } = require('../config/constants')
 
 const budgeting = module.exports = (() => {
     const db = require('../models')
     const clientManagement = require('./clientManagement')
 
+    const checkExistingPaymentsWithExternalId = async (params) => {
+        const existingPayments = await db.models.Payment.findAll({
+            where: {
+                external_uuid: params.id
+            }
+        })
+        
+        if (existingPayments) {
+            existingPayments.forEach(async payment => {
+                const client = await clientManagement.findClientWithId(payment.client_id)
+
+                if (client.currency != DEFAULT_STRIPE_CURRENCY) {
```

Comment:
This comparison should be between `client.currency` and `stripeInvoice.currency`